### ISSUE TITLE
Unable to import models which use uuid primary key

### ISSUE THREAD

**Issue by matthewelwell** [django-import-export/django-import-export#1274]
**Describe the bug**
Importing models which use a uuid primary key (or any value that is generated in python, not the database) fails silently and the models do not import.

**To Reproduce**
1. Create a model with a uuid primary key (`id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)`)
2. Create a custom `ModelResource` class to import and set `use_bulk` = True
3. Import a file containing NEW records
4. The import appears to be successful but the records are not imported

**Versions (please complete the following information):**
 - Django Import Export: 2.5.0
 - Python 3.9
 - Django 3.2

**Expected behavior**
The records should be imported. 

**Screenshots**
N/a

**Additional context**
I believe it's related to the following code in the `save_instance` method of the `import_export.resources.Resource` class: 

```python
          if self._meta.use_bulk:
            if instance.pk:
                self.update_instances.append(instance)
            else:
                self.create_instances.append(instance)
```

Since, when using a UUID, the pk is generated before the object is saved to the database, checking the value of `pk` against the instance is not a valid way to confirm whether the object exists or not. 

A simple (if somewhat crude) fix would be, I believe: 

```python
          if self._meta.use_bulk:
            creating = (
                (hasattr(instance, "_state") and getattr(instance._state, "adding"))
                or instance.pk is None
            )
            if not creating:
                self.update_instances.append(instance)
            else:
                self.create_instances.append(instance)
```

**Comment by matthewhegarty** [comment]
HI @matthewelwell - thanks for raising this.

I've added a simple fix in which the flag for determining create / update is taken from an [earlier step](https://django-import-export.readthedocs.io/en/latest/api_resources.html#import_export.resources.Resource.get_or_init_instance) which has already attempted to load the existing object.  This is effectively the same as it was, except it doesn't rely on `pk`.

I'd be grateful if you could test that this would fix your issue.

**Comment by matthewelwell** [comment]
Hey @matthewhegarty that sounds like a much better solution than the one I posted above. 

I've temporarily forked the code and implemented my solution above to get things working. I will test this next week and let you know 👍

**Comment by matthewhegarty** [comment]
@matthewelwell A couple of issues were raised in code review, so I'm going to look again at this.

**Comment by matthewhegarty** [comment]
@matthewelwell I'm now taking a different approach.  I'd be very grateful if you could review the [changes](#1278) and let me know if they will work for you.

**Comment by matthewelwell** [comment]
Thanks @matthewhegarty, the current change looks like you've gone pretty much back to what it was originally (which would work for a simpleton such as myself who is not overriding anything 😄 ). Is this the final settled approach?

My concern around the `pk_attr` approach that was discussed is that (although this may be useful in other scenarios) it wouldn't work here since the problem is not around the field name, it's around the fact that the value is set on the instantiation of the model object, not at the database level as with a usual auto incrementing ID.

Anyway, I've tested your branch (using commit sha 21390b7) and it works perfectly for my use case.

**Comment by matthewhegarty** [comment]
Hi @matthewelwell, Yes I had to go back to the original approach due to the realisation that it wouldn't actually fix your issue!  This will be the final settled approach, but we're likely delaying it to v3 (release date unknown) due to it breaking the API.

Obviously you'll have to port it to your fork, or use the branch until v3 is released (which I doubt will be soon).

Thanks again for raising this and testing!

**Comment by matthewhegarty** [comment]
Fixed by #1274

**Comment by adaikalarajd** [comment]
Hi @matthewhegarty, As you mentioned this issue fixes will be released in V3, any luck to know about the V3 release date, Thank you for your support.
